Tasks/VsTest/vstest.ts

```diff
@@ -343,8 +350,249 @@ function getTestResultsDirectory(settingsFile: string, defaultResultsDirectory:
     return defer.promise;
 }
 
+function getTICollectorURI () {
+    return "datacollector://microsoft/TestImpact/1.0";
+}
+
+function getTIAssemblyQualifiedName(vsVersion) {
+    return "Microsoft.VisualStudio.TraceCollector.TestImpactDataCollector, Microsoft.VisualStudio.TraceCollector, Version=" + vsVersion + ".0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a";
+}
+
+function getTIFriendlyName() {
+    return "Test Impact";
+}
+
+function getTestImpactAttributes(vsVersion) {
+    return { uri: getTICollectorURI(),
+             assemblyQualifiedName: getTIAssemblyQualifiedName(vsVersion),
+             friendlyName: getTIFriendlyName() };
+}
+
+function isTestImapctCollectorPresent(dataCollectorArray) {
+    var found = false;
+    var tiaFriendlyName = getTIFriendlyName().toUpperCase();
+    for (var i = 0; i < dataCollectorArray.length; i++) {
+        try {
```

Comment:
then add that case in if condition. you are not supposed to catch exceptions for properties checking

---
In reply to: [68179179](https://github.com/Microsoft/vsts-tasks/pull/1932#discussion_r68179179) [](ancestors = 68179179)